community/kernel/src/test/java/org/neo4j/graphdb/IndexingCompositeQueryAcceptanceTest.java

```diff
@@ -0,0 +1,316 @@
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.graphdb;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TestName;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.neo4j.collection.primitive.Primitive;
+import org.neo4j.collection.primitive.PrimitiveLongIterator;
+import org.neo4j.collection.primitive.PrimitiveLongSet;
+import org.neo4j.test.mockito.matcher.Neo4jMatchers;
+import org.neo4j.test.rule.ImpermanentDatabaseRule;
+
+import static org.hamcrest.core.IsEqual.equalTo;
+import static org.junit.Assert.assertThat;
+import static org.neo4j.helpers.collection.Iterators.array;
+
+@RunWith( Parameterized.class )
+public class IndexingCompositeQueryAcceptanceTest
+{
+    @ClassRule
+    public static ImpermanentDatabaseRule dbRule = new ImpermanentDatabaseRule();
+    @Rule
+    public final TestName testName = new TestName();
+
+    @Parameterized.Parameters
+    public static List<Object[]> data()
+    {
+        return Arrays.asList(
+                testCase( array( 2, 3 ), biIndexSeek, true ),
+                testCase( array( 2, 3 ), biIndexSeek, false ),
+                testCase( array( 2, 3, 4 ), triIndexSeek, true ),
+                testCase( array( 2, 3, 4 ), triIndexSeek, false ),
+                testCase( array( 2, 3, 4, 5, 6 ), mapIndexSeek, true ),
+                testCase( array( 2, 3, 4, 5, 6 ), mapIndexSeek, false )
+        );
+    }
+
+    @Parameterized.Parameter( 0 )
+    public String[] keys;
+    @Parameterized.Parameter( 1 )
+    public Object[] values;
+    @Parameterized.Parameter( 2 )
+    public Object[][] nonMatching;
+    @Parameterized.Parameter( 3 )
+    public IndexSeek indexSeek;
+    @Parameterized.Parameter( 4 )
+    public boolean withIndex;
+
+    private static Label LABEL = Label.label( "LABEL1" );
+    private GraphDatabaseService db;
+
+    @Before
+    public void setup()
+    {
+        db = dbRule.getGraphDatabaseAPI();
+        if ( withIndex )
+        {
+            Neo4jMatchers.createIndex( db, LABEL, keys );
```

Comment:
why on <strike>Earth</strike> any planet! we will gonna have a method on Neo4jMatchers! that does index creation!

Reply:
Yes... that was surprising to me too. Changed it to use Core API in the new test classes.

Reply:
I traced this to a commit by Ian in 2013. Seems to be part of a big push in expanding the testing framework. And it is used all over the place. I think cleaning this up is an entirely separate job, not for this PR.